source/MaterialXRuntime/RtLook.h

```diff
@@ -88,6 +88,9 @@ class RtMaterialAssign : public RtTypedSchema
     RtAttribute getExclusive() const;
 };
 
+/// Function to be called to throw an exception if element is being created from within a compound
+void throwIfNotTopLevelElement(const RtToken& typeName, const RtPath& parentPath);
```

Comment:
This function does not need to be exposed in the public API. Maybe put it as a path utility function, in PvtPath.h, something like: 

`void throwIfNotRoot(const RtPath& path, const std::string& msg);`

(where msg is the error message to throw so this can be changed depending on use-case)